### PR TITLE
Update haxe.display

### PR DESCRIPTION
**Pull request by Simn**
Have to sync this with vshaxe. This PR doesn't include the json-rpc protocol definitions because those depend on jsonrpc.Types. I suppose they can remain in vshaxe for now and maybe be externalized later.

PR also includes some adjustments to the server tests to demonstrate a use-case for these type structures.

### CODE REVIEW

File: std/haxe/display/Display.hx [hunk 1]
```diff
@@ -0,0 +1,459 @@
...
+	var Abstract = "TAbstract";
+	var AbstractField = "TAbstractField";
+	var Enum = "TEnum";
+	var Typedef = "TTypedef";
+	var AnyField = "TAnyField";
+	var Expr = "TExpr";
+	var TypeParameter = "TTypeParameter";
+}
+
+enum abstract Platform(String) {
```
Comment (RealyUniqueName) [inline std/haxe/display/Display.hx:270]: Shouldn't this be `Target`? Also related to other "platform" symbols here

File: std/haxe/display/Display.hx [hunk 2]
```diff
@@ -0,0 +1,459 @@
...
+ * DEALINGS IN THE SOFTWARE.
+ */
+
+package haxe.display;
+
+import haxe.display.JsonModuleTypes;
+import haxe.display.Protocol;
+import haxe.display.Position;
+
+enum abstract CompilerMetadata(String) to String {
```
Comment (Gama11) [inline std/haxe/display/Display.hx]: I don't think this needs to be here.

File: std/haxe/display/Display.hx (hunk 1 above)
Comment (Simn) [reply std/haxe/display/Display.hx:270]: Actually we should sync the entire metadata section with the new src-json/meta.json content.

Comment (Gama11) [reply std/haxe/display/Display.hx:270]: The problem is that metadata already has a `target`, but with a different meaning.

Comment (Aurel300) [reply std/haxe/display/Display.hx:270]: We can rename that one to `appliesTo` maybe?

**Comment by Gama11** [comment]
>This PR doesn't include the json-rpc protocol definitions because those depend on jsonrpc.Types.

Not sure it's a good idea to split them up, they kind of belong together. I think it would just come down to making `HaxeRequestMethod` `HaxeNotificationMethod` abstracts themselves, instead of typedefs to their jsonrpc counterparts?

**Comment by Simn** [comment]
I'm not sure. The type definitions can exist without the json-rpc methods, which are built on top of them. In that sense it's fair that the type definitions are in std, while the json-rpc parts are not.

Not sure what you mean regarding abstracts.

File: std/haxe/display/Display.hx (hunk 1 above)
Comment (RealyUniqueName) [reply std/haxe/display/Display.hx:270]: compilationTarget, metaTarget

**Comment by Gama11** [comment]
Well, `RequestMethod` / `NotificationMethod` are the only things needed from the jsonrpc lib, and there's really not a lot to them: https://github.com/vshaxe/vscode-json-rpc/blob/master/src/jsonrpc/Types.hx#L101-L109

So the `Haxe` types could be turned into abstracts themselves instead of being typedefs to those.

**Comment by Simn** [comment]
Right, but they still belong to jsonrpc more than they belong to haxe.display. But maybe I'm overthinking this.

**Comment by Gama11** [comment]
It's really just a trick to attach some type parameters to a String. And that String definitely belongs to haxe.display, because you need to know what the identifier of the method you want to call is.

What you're doing instead right now is hardcoding that string in the test (`"display/completion"`).

**Comment by Simn** [comment]
Ok I misunderstood what you meant. We can turn `HaxeRequestMethod` into an abstract and then it fits nicely indeed.

**Comment by Simn** [comment]
done

**Pull request merged**